sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala

```diff
@@ -116,47 +116,53 @@ object OrcUtils extends Logging {
   }
 
   /**
-   * Returns the requested column ids from the given ORC file. Column id can be -1, which means the
-   * requested column doesn't exist in the ORC file. Returns None if the given ORC file is empty.
+   * @return Returns the requested column ids from the given ORC file and Boolean flag to use actual
+   * schema or result schema. Column id can be -1, which means the requested column doesn't
+   * exist in the ORC file. Returns None if the given ORC file is empty.
    */
   def requestedColumnIds(
       isCaseSensitive: Boolean,
       dataSchema: StructType,
       requiredSchema: StructType,
       reader: Reader,
-      conf: Configuration): Option[Array[Int]] = {
+      conf: Configuration): (Option[Array[Int]], Boolean) = {
```

Comment:
I think the return type should be `Option[(Array[Int], Boolean)]`

Reply:
I did similar change by returning the one attribute in the return type. I was asked to make it two attributes to return from this method
Please find the review comment for that  - https://github.com/apache/spark/pull/29045#discussion_r453335630

So I change it after this review comment

Reply:
how about
```
def requestedColumnIds(...): Option[(Array[Int], Boolean)]
...
val result = ... { OrcUtils.requestedColumnIds... }
if (result.isEmpty) {
  Iterator.empty
} else {
  val (requestedColIds, canPruneCols) = result.get
  ...
}
```

Reply:
done